saleor/graphql/page/filters.py

```diff
@@ -212,16 +222,130 @@ def filter_pages_by_attributes(qs, value):
             attr_filter_expression &= filter_by_date_time_attribute(
                 attr.id, attr_value["date_time"], qs.db
             )
+        elif attr.input_type == AttributeInputType.REFERENCE:
+            attr_filter_expression &= filter_pages_by_reference_attributes(
+                attr.id, attr_value["reference"], qs.db
+            )
     if attr_filter_expression != Q():
         return qs.filter(attr_filter_expression)
     return qs.none()
 
 
+def filter_pages_by_reference_attributes(
+    attr_id: int | None,
+    attr_value: dict[
+        Literal[
+            "referenced_ids", "page_slugs", "product_slugs", "product_variant_skus"
+        ],
+        CONTAINS_TYPING,
+    ],
+    db_connection_name: str,
+):
+    filter_expression = Q()
+    if "referenced_ids" in attr_value:
+        filter_expression &= filter_by_contains_referenced_object_ids(
+            attr_id,
+            attr_value["referenced_ids"],
+            db_connection_name,
+            assigned_attr_model=AssignedPageAttributeValue,
+            assigned_id_field_name="page_id",
+        )
+    if "page_slugs" in attr_value:
+        filter_expression &= filter_by_contains_referenced_pages(
+            attr_id,
+            attr_value["page_slugs"],
+            db_connection_name,
+            identifier_field_name="slug",
+            assigned_attr_model=AssignedPageAttributeValue,
+            assigned_id_field_name="page_id",
+        )
+    if "product_slugs" in attr_value:
+        filter_expression &= filter_by_contains_referenced_products(
+            attr_id,
+            attr_value["product_slugs"],
+            db_connection_name,
+            identifier_field_name="slug",
+            assigned_attr_model=AssignedPageAttributeValue,
+            assigned_id_field_name="page_id",
+        )
+    if "product_variant_skus" in attr_value:
+        filter_expression &= filter_by_contains_referenced_variants(
+            attr_id,
+            attr_value["product_variant_skus"],
+            db_connection_name,
+            identifier_field_name="sku",
+            assigned_attr_model=AssignedPageAttributeValue,
+            assigned_id_field_name="page_id",
+        )
```

Comment:
You are providing almost the same arguments to those functions, you can define the defaults in variable and unpack them when calling the method to improve readability and enhance what is changing
```suggestion
    kwargs = {
          "attr_id": attr_id,
          "db_connection_name": db_connection_name,
          "assigned_attr_model": AssignedPageAttributeValue,
          "assigned_id_field_name": "page_id",
     }
    if "page_slugs" in attr_value:
        filter_expression &= filter_by_contains_referenced_pages(
            attr_value=attr_value["page_slugs"],
            identifier_field_name="slug",
            **kwargs
        )
    if "product_slugs" in attr_value:
        filter_expression &= filter_by_contains_referenced_products(
            attr_value["product_slugs"],
            identifier_field_name="slug",
            **kwargs
        )
    if "product_variant_skus" in attr_value:
        filter_expression &= filter_by_contains_referenced_variants(
            attr_value["product_variant_skus"],
            identifier_field_name="sku",
            **kwargs
        )
```

Reply:
I tried to do that, but this will also break the mypy. As passing **kwars, causes that mypy is not able to match the typing correctly. I've ended up with the same structure, as I assumed that it is better to have proper typing than few lines less